File: invision-backend/app/Enums/OrderStatus.php
<?php

namespace App\Enums;

enum OrderStatus: string
{
    case Draft = 'draft';
    case Confirmed = 'confirmed';
    case Delivered = 'delivered';
    case Cancelled = 'cancelled';
    case Returned = 'returned';

    public function label(): string
    {
        return match ($this) {
            self::Draft => 'Draft',
            self::Confirmed => 'Confirmed',
            self::Delivered => 'Delivered',
            self::Cancelled => 'Cancelled',
            self::Returned => 'Returned',
        };
    }

    public function color(): string
    {
        return match ($this) {
            self::Draft => 'gray',
            self::Confirmed => 'blue',
            self::Delivered => 'green',
            self::Cancelled => 'red',
            self::Returned => 'yellow',
        };
    }
}

File: invision-mobile/lib/src/core/enums/order_status.dart
enum OrderStatus {
  draft,
  confirmed,
  delivered,
  cancelled,
  returned;

  String get label => switch (this) {
    draft => 'Draft',
    confirmed => 'Confirmed',
    delivered => 'Delivered',
    cancelled => 'Cancelled',
    returned => 'Returned',
  };

  static OrderStatus fromString(String value) {
    return switch (value) {
      'draft' => OrderStatus.draft,
      'confirmed' => OrderStatus.confirmed,
      'delivered' => OrderStatus.delivered,
      'cancelled' => OrderStatus.cancelled,
      'returned' => OrderStatus.returned,
      _ => OrderStatus.draft,
    };
  }
}
